include/onnxruntime/core/graph/graph.h

```diff
@@ -342,52 +342,45 @@ class Node {
   /** Gets the number of output edges from this Node */
   size_t GetOutputEdgesCount() const noexcept { return relationships_.output_edges.size(); }
 
-  /** Add an attribute to this Node with specified attribute name and value. */
-  void AddAttribute(std::string attr_name, const ONNX_NAMESPACE::AttributeProto& value);
-  void AddAttribute(std::string attr_name, ONNX_NAMESPACE::AttributeProto&& value);
+  /** Adds an AttributeProto to this Node.
+  @remarks The attribute name is used as the key in the attribute map. */
+  void AddAttributeProto(ONNX_NAMESPACE::AttributeProto value);
 
-#define ADD_ATTR_INTERFACES(TypeName)                              \
-  void AddAttribute(std::string attr_name, const TypeName& value); \
-  void AddAttribute(std::string attr_name,                         \
-                    gsl::span<TypeName const> values);
+  /** Adds an attribute to this Node with the specified attribute name and value(s). */
+  void AddAttribute(std::string attr_name, int64_t value);
+  void AddAttribute(std::string attr_name, gsl::span<const int64_t> values);
 
```

Comment:
Why do we do int64_t separately instead of via ADD_ATTR_INTERFACES?

Reply:
thought it was easier to document a function declaration than the macro that generates the declarations

Reply:
maybe add an `@remark` calling out that's why, and mentioning this needs to be kept in sync with the ADD_ATTR_INTERFACES macro

Reply:
sounds good, added a comment